src/main/java/spoon/support/StandardEnvironment.java

```diff
@@ -256,13 +256,20 @@ public void report(Processor<?> processor, Level level, CtElement element, Strin
 		if (sp == null) {
 			buffer.append(" (Unknown Source)");
 		} else {
-			// TODO: will explode if type == null
-			buffer.append(" at " + type.getQualifiedName() + ".");
+			if (type != null) {
+				buffer.append(" at " + type.getQualifiedName() + ".");
```

Comment:
```suggestion
				buffer.append(" at ").append(type.getQualifiedName()).append(".");
```
If you already have a buffer I'd use its `append` method instead of string concatenation inside the argument.